src/app/core/campaign/campaign-store.service.ts

```diff
@@ -55,6 +57,100 @@ export class CampaignStoreService {
     }
   }
 
+  getFlightAvailabilityRollup$(flightId: string): Observable<Availability[]> {
+    return this.campaign$.pipe(
+      map(state => {
+        // availability of current flights
+        const availabilityZones = state.flights[flightId] &&
+          state.flights[flightId].localFlight &&
+          state.flights[flightId].localFlight.zones &&
+          state.availability &&
+          state.flights[flightId].localFlight.zones
+            .filter(zone => state.availability[`${flightId}-${zone}`])
+            .map(zone => state.availability[`${flightId}-${zone}`]);
+        // each zone
+        const zoneWeeks = availabilityZones && availabilityZones.map((availability: Availability) => {
+          let weekBeginString: string;
+          let weekEnd: Date;
+          // acc weeks
+          return availability.totals.groups.reduce((acc, day) => {
+            const dayDate = new Date(day.startDate + ' 0:0:0');
+            if (!weekEnd || weekEnd.valueOf() <= dayDate.valueOf()) {
```

Comment:
I can't readily see how this conditional would ever be `false`.

Reply:
It breaks the weeks up when they cross over the ending of the previous week. It's true when `weekEnd` is in the previous week, and `dayDate` is day in the following week. It's false when `dayDate` is in the current week. When false, it accumulates onto the acc.totals.groups[weekBeginString]. This could probably be better. We'll be doing more work here I suppose.